### PR TITLE
Add variation inventory quick actions

### LINKED ISSUE #39789: Add Inventory item to the Quick Actions menu per Variation item
**Prerequisites (mark completed items with an [x]):**
- [ ] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [ ] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**

User needs a way to modify each variation quickly under the Variations tab.

**Epic**

#35670

**ACs**

- [ ] Add `Inventory` item to the dropdown menu AkVNGImLgSqCObTQ3idVn7-fi-1761_310147
- [ ] Add the dropdown submenu from the `Inventory` item AkVNGImLgSqCObTQ3idVn7-fi-1641_215678
- [ ] Add `Update stock` item to the dropdown menu AkVNGImLgSqCObTQ3idVn7-fi-1761_310346
- [ ] Add `Toggle “Track quantity”` item to the dropdown menu AkVNGImLgSqCObTQ3idVn7-fi-1761_310324
- [ ] Add `Set status to In stock` item to the dropdown menu AkVNGImLgSqCObTQ3idVn7-fi-1761_310302
- [ ] Add `Set status to Out of stock` item to the dropdown menu AkVNGImLgSqCObTQ3idVn7-fi-1761_310280
- [ ] Add `Set status to On back order` item to the dropdown menu AkVNGImLgSqCObTQ3idVn7-fi-1761_310258
- [ ] Add `Edit low stock threshold` item to the dropdown menu AkVNGImLgSqCObTQ3idVn7-fi-1761_310236
- [ ] When the user click on each item a modal (`window.prompt`) should be shown to ask for the new vale AkVNGImLgSqCObTQ3idVn7-fi-1641_218691
- [ ] Save the values of the selected variation
- [ ] Show a snackbar informing users that the variation were updated AkVNGImLgSqCObTQ3idVn7-fi-1641_218705
- [ ] Track all the quick actions behavior, including the modal to add/update values

### PR DESCRIPTION
**Pull request by louwie17**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR adds inventory related quick actions to the variations list.

Closes #39789 

Screenshot 2023-08-28 at 1 57 52 PM

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**Requires feature flag: `product-variation-management`**

1. Load this branch and enable the new product editor under **WooCommerce > Settings > Advanced > Features**
2. Install the WooCommerce Beta tester plugin and enable the `product-variation-management` feature under **Tools > WCA Test Helper > Features**
3. Go to **Products > Add New** and create a new product, save it, and add multiple variation options under Variations tab.
4. Once the variations are generated, notice a 3 dot menu, click on this, it should contain an **Inventory** option.
5. Go through each option as mentioned below, to test if it got updated correctly you could click **Preview** and select the specific variation ( this will require you to add prices first ).
     1. **Update stock** - should show a window prompt asking for stock and set `manage_stock` to true automatically ( also check if cancelling the prompt doesn't update the value )
     2. **Toggle "track quantity"** should toggle tracking quantity ( if enabled it should show the quantity within the preview for the specific variation ).
     3. **Set status to in stock** will set status to instock and disable the manage stock
     4. **Set status to out of stock** will set status to outofstock and disable the manage stock
     5. **Set status to on back order** will set status to on back order and disable the manage stock
     6. **Edit low stock threshold** should set the low stock threshold and auto enable the manage stock ( best way to check this is disabling the **new product editor** and check the old editor low stock threshold value.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

**Review by mdperez86 — APPROVED** [review]
LGTM nice job @louwie17 !!! It would be good if we could move the Pricing submenu to its own component. I know it's not part of this PR but, you can do it here if you have time.

**Comment by louwie17** [comment]
> It would be good if we could move the Pricing submenu to its own component. I know it's not part of this PR but, you can do it here if you have time.

@mdperez86 done, if you want to have another look? :)

File: packages/js/product-editor/src/components/variations-table/variation-actions-menu/variation-actions-menu.tsx [hunk 1]
```diff
@@ -67,14 +25,16 @@ export function VariationActionsMenu( {
 	function handlePrompt(
 		propertyName: keyof ProductVariation,
 		label: string = __( 'Enter a value', 'woocommerce' ),
-		parser: ( value: string ) => unknown = ( value ) => value
+		parser: ( value: string ) => unknown = ( value ) => value,
+		additionalProps: Partial< ProductVariation > = {}
```
Comment (mdperez86) [inline packages/js/product-editor/src/components/variations-table/variation-actions-menu/variation-actions-menu.tsx]: I think that if the `parser` function returns the whole variation we don't need this `additionalProps` anymore right. Also the `handlePrompt` function does not need to know about the `propertyName` in that case. I was having a similar thinking doing Shipping https://github.com/woocommerce/woocommerce/pull/39925/files#diff-2f82b22147fa0e6b6dd8a3d351ea8856cbdfbd185ee4d79ca54c147373d0c0ccR64 and I thought changing it gave us more flexibility, for example, your case here.

Before
```
onChange( {
	[ propertyName ]: parser( value.trim() ),
	...additionalProps,
} );
```

After

```
onChange( parser( value.trim() ) );
```

**Review by mdperez86 — APPROVED** [review]
@louwie17  I left one non small comment for you here @louwie17 you can do it in this PR or not. 
This is approved anyway.
Great job here!!!!!

File: packages/js/product-editor/src/components/variations-table/variation-actions-menu/variation-actions-menu.tsx (hunk 1 above)
Comment (louwie17) [reply packages/js/product-editor/src/components/variations-table/variation-actions-menu/variation-actions-menu.tsx]: @mdperez86 updated as part of https://github.com/woocommerce/woocommerce/pull/39935/commits/772842f4f51fb55dc4268cabfecbcb5b03194e7e

**Pull request merged**